tree/ntuple/v7/inc/ROOT/RDaos.hxx

```diff
@@ -149,18 +151,52 @@ public:
    using DistributionKey_t = RDaosObject::DistributionKey_t;
    using AttributeKey_t = RDaosObject::AttributeKey_t;
    using ObjClassId_t = RDaosObject::ObjClassId;
-  
+
+   /// \brief A pair of <object ID, distribution key> that can be used to issue a fetch/update request for multiple
+   /// attribute keys.
+   struct ROidDkeyPair {
+      daos_obj_id_t oid{};
+      DistributionKey_t dkey{};
+      ROidDkeyPair() = default;
+
+      inline bool operator==(const ROidDkeyPair &other) const
+      {
+         return this->oid.lo == other.oid.lo && this->oid.hi == other.oid.hi && this->dkey == other.dkey;
+      }
+
+      struct Hash {
+         auto operator()(const ROidDkeyPair &x) const
+         {
+            /// Implementation borrowed from `boost::hash_combine`. Comparable to initial seeding with `oid.hi` followed by two
+            /// subsequent hash calls for `oid.lo` and `dkey`.
+            auto seed = std::hash<uint64_t>{}(x.oid.hi);
+            seed ^= std::hash<uint64_t>{}(x.oid.lo) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+            seed ^= std::hash<DistributionKey_t>{}(x.dkey) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+            return seed;
+         }
+      };
+   };
+
    /// \brief Describes a read/write operation on multiple objects; see the `ReadV`/`WriteV` functions.
    struct RWOperation {
       RWOperation() = default;
-      RWOperation(daos_obj_id_t o, DistributionKey_t d, AttributeKey_t a, std::vector<d_iov_t> &v)
-         : fOid(o), fDistributionKey(d), fAttributeKey(a), fIovs(v) {};
+      RWOperation(daos_obj_id_t o, DistributionKey_t d, std::vector<AttributeKey_t> &&as, std::vector<d_iov_t> &&vs)
+         : fOid(o), fDistributionKey(d), fAttributeKeys(std::move(as)), fIovs(std::move(vs)){};
```

Comment:
I think that this constructor is currently unused.  However, given that it is intended to provide the initial state (after which, `insert()` can be called), I would suggest to change it so that it doesn't adopt the storage of the provided vectors.
```suggestion
      RWOperation(daos_obj_id_t o, DistributionKey_t d, const std::vector<AttributeKey_t> &as, const std::vector<d_iov_t> &vs)
         : fOid(o), fDistributionKey(d), fAttributeKeys(as), fIovs(vs) {};
```